docs/querying/tips-good-queries.md

```diff
@@ -0,0 +1,198 @@
+
+You can follow the tutorial [Using TopN approximation in Druid queries](https://github.com/apache/druid/blob/master/examples/quickstart/jupyter-notebooks/notebooks/03-query/02-approxRanking.ipynb) to work through some examples with approximation turned on and off. The tutorial [Get to know query view](../tutorials/tutorial-sql-query-view.md) demonstrates running aggregate queries in the Druid console.
+
+### Manually tune your queries
+
+This section outlines techniques you can use to improve your query accuracy and performance.
+
+#### Query one table at a time
+
+Query a single table at a time to minimize the load on the Druid processor.
+
+#### Select specific columns
+
+Only select the columns needed for the query instead of retrieving all columns from the table. This reduces the amount of data retrieved from the database, which improves query performance.
+
+#### Use filters
+
+Use filters, for example the WHERE clause, and filter on time. Try to minimize the use of non-equal filters, because they're very resource-intensive.
+
+The following example query filters on `__time` and `product`:
+
+```
+SELECT
+  FLOOR(__time to day),
+  product,
+  sum(quantity * price) as revenue
+FROM "orders"
+WHERE
+  __time > '2023-08-20' and product = 'product 1'
+GROUP BY 1, 2
+```
+
+The following example uses a wildcard filter on the `diffUrl` column:
+
+```
+SELECT * from Wikipedia
+WHERE diffUrl LIKE 'https://en.wikipedia%'
+AND __time BETWEEN '2016-06-27 00:00' AND '2016-06-27 12:00'
+```
+
+#### Shorten your queries
+
+Make your queries shorter where possible&mdash;Druid processes shorter queries faster. You might also be able to divide a single query into multiple queries.
+
+For example, the following query aggregates over multiple datasources using UNION ALL:
+
+```
+SELECT id, SUM(revenue) FROM
+   (SELECT id, revenue from datasource_1
+UNION ALL
+  SELECT id, revenue FROM datasource_2)
+...
+UNION ALL
+   SELECT id, revenue FROM datasource_n)
+GROUP BY id
+```
+
+To simply this query, you could split it into several queries and then aggregate the results. For example:
+
+```
+SELECT id, SUM(revenue) FROM datasource_1
+
+SELECT id, SUM(revenue) FROM datasource_2
+...
+SELECT id, SUM(revenue) FROM datasource_n
+```
+
+#### Minimize or remove subqueries
+
+Consider whether you can pre-compute a subquery task and store it as a join or make it a part of the datasource.
+
+#### Consider alternatives to GroupBy
+
+Consider using Timeseries and TopN as alternatives to GroupBy. See [GroupBy queries: alternatives](./groupbyquery.md#alternatives) for more information.
+
+Avoid grouping on a high cardinality column, for example user ID. Investigate whether you can filter on user ID first. Find out whether your dataset is already partitioned by user ID. See [Partitioning](../ingestion/partitioning.md) for information on configuring paritioning.
```

Comment:
This paragraph reads a bit odd. How do the points relate -- what is the takeaway if you CAN filter on user ID. If your dataset is partitioned, does that mean you can group on it?

Also typo in "partitioning" in the end of the sentence

Reply:
I don't think this part made much sense - removed the reference to partitioning here.